src/control/lib/control/system.go

```diff
@@ -231,6 +202,46 @@ func SystemNotify(ctx context.Context, req *SystemNotifyReq) (*SystemNotifyResp,
 	return resp, convertMSResponse(ur, resp)
 }
 
+// EventForwarder implements the events.Handler interface, increments sequence
+// number for each event forwarded and distributes requests to MS access points.
+type EventForwarder struct {
+	seq       uint64
```

Comment:
I'm not sure what the utility of this sequence number is. It's not persisted anywhere so it will reset when the server restarts. Is that going to confuse anything that is depending on the sequence always incrementing?

Reply:
the idea was to be able to match a response to a given request (ack) as there isn't always going to be another way to identify that an event has been handled

Reply:
So I guess in that case it doesn't matter if the sequence number is reset because it's only relevant to the sender. I assume that the goal would be to retry or something if a corresponding response isn't received in some amount of time? If so, I think I would prefer to keep the retry logic in the MS client rather than adding another layer here.

I think it's fine to leave the sequence number in for now though. We may find another use for it (e.g. distinguishing between N similar events with distinct sequence numbers and N duplicates of the same event).